libs/lib-client/diagram/src/lib/drag-tiles/drag-tile.service.ts

```diff
@@ -77,4 +77,43 @@ export class DragTileService {
       parentId,
     };
   }
+
+  getAllRowsParentTiles(tileMatrix, rowIndexes) {
```

Comment:
We ask consuming component to call this method with Graph Nodes and we can initialise an empty map or a map<string, rowIndexes: number[]> if getRowIndexOf is provided as a parameter
```suggestion
  initializeRelationsLookupTable(nodes: NodeDictionary, getRowIndexOf?: (string) => number) {
```

Reply:
Agree, but a resource will not consume this method if it cannot have multiple rows. So i think this condition is not necessary 🤔

Reply:
IMO, Every resource must call this initialization service method. Else, we can miss an aspect of resetting the data. Think about it and plan accordingly